src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -99,31 +100,41 @@ public Task StopAsync(CancellationToken cancellationToken)
             return Task.CompletedTask;
         }
 
-        public IDisposable SuspendRestart()
+        public IDisposable SuspendRestart(bool autoRestart)
         {
-            return new SuspendRestartRequest(this);
+            return new SuspendRestartRequest(this, autoRestart);
         }
 
-        public void ResumeRestart()
+        private async Task RestartIfScheduledAsync()
         {
-            Interlocked.Decrement(ref _suspensionRequestsCount);
-            _typedLogger.LogDebug($"Resuming restart. ({_suspensionRequestsCount} requests).");
+            if (_restartScheduled)
```

Comment:
Seems like we don't need the complexity of this new boolean. You can just use the existing _suspensionRequestsCount and in the Disposal, if this is going from 1 to 0 you know there were suspended restarts that can then be executed if autoResume is true.

Reply:
We need to restart only if there was a file change for which a restart is needed. Checking _suspensionRequestsCount does not work, there can be a PUT which does not change the files.